lib/svg.dart

```diff
@@ -1,51 +1,50 @@
 import 'dart:ui';
+import 'package:flame/extensions/vector2.dart';
 import 'package:flutter_svg/flutter_svg.dart';
 
 import 'package:flame/flame.dart';
-import 'package:flame/position.dart';
 
 class Svg {
   DrawableRoot svgRoot;
   Size size;
+  
+  Svg(this.svgRoot);
 
-  Svg(String fileName) {
-    Flame.assets.readFile(fileName).then((svgString) async {
-      svgRoot = await svg.fromSvgString(svgString, svgString);
+  static Future<Svg> load(String fileName) async {
+    return Flame.assets.readFile(fileName).then((svgString) async {
```

Comment:
I thought you said we didn't need the extension yesterday?
Hmm, would the flame game cache be cleared too if a Navigator pulled in another widget instead of the flame game?

Reply:
Sorry, maybe my message was confusing, my comment was about not needing the new collection 'svgs', but still have the 'loadSvg' method added on game, so that method could use the assets instance.

I am not entirely sure how the navigator works, but I think yes, when you navigate to other screens, it is removed from the tree

Reply:
If that is the case that it is that volatile, don't we want everything in the global cache?
Otherwise all games that are built with pure flutter screens together with GameWidget will have to reload all assets every time they go back to a screen with the GameWidget?

Reply:
Yes, and I think this is ok, not having an automatic way of handling caching was one of the things that we got complains in the past, that is why we did that new approach on Game having a cache of its own, and cleaning it.

But we also kept the global cache available (`Flame.images` and `Flame.assets`), because maybe in some games, reloading all the assets could be a hassle, so what I am suggesting here is to have an  easy way of loading with the game instances, just like we have for sprites and spriteanimations, but the developer still have the option to load things using the global cache.

Reply:
What is the argument for not having the global cache as default?
Wouldn't the cache be cleared at every little update to the widget tree now?

Reply:
> What is the argument for not having the global cache as default?

The argument is what I just said above, there was not an automatic handling of cache on Flame, now when using the instances from the game, it is cleared when the gamewidget is removed from the tree.

> Wouldn't the cache be cleared at every little update to the widget tree now?

No 🤔, only when the game widget is removed from the widget tree.

What I think we could do, that could even be an improvement that we could expand to the Sprite and SpriteAnimation on the core of Flame is doing something this:

```dart
 // Keep the load method, but we have the assets as an optional paramenter with the default to the global
static Future<Svg> load(String fileName, { AssetsCache assets = Flame.assets }) async {
    assets.readFile(fileName).then((svgString) async {
    	    return Svg(await svg.fromSvgString(svgString, svgString));
    });
}
 
extension SvgLoader on Game {
  Future<Svg> loadSvg(String fileName) => Svg.load(fileName, assets: assets)
}
```

That way both global, and game caching are quite easy to use

Reply:
I did some tests and it is not at all as volatile as I'd thought, so that is good. :)

I have added the extension and the optional cache argument too now.

Reply:
Good, I think I will do that optional think on Flame as well!

Reply:
Forgot to push, now it is up.